code/__DATASTRUCTURES/priority_queue.dm

```diff
@@ -0,0 +1,65 @@
+
+//////////////////////
+//PriorityQueue object
+//////////////////////
+
+//an ordered list, using the cmp proc to weight the list elements
+/PriorityQueue
+	var/list/L //the actual queue
```

Comment:
Ideally a queue would use a linked list for the benefits of being able to remove/add the front element with O(1) efficiency.

Not sure how that would work in BYOND, you will need to do benchmarks. I am also assuming BYOND lists are dynamic arrays, they could be linked lists for all I know.